### PR TITLE
immer git

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] change git to immer
- [x] use chainaction2
- [x] use some hooks

### CODE REVIEW

File: shared/.eslintrc.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ module.exports = {
     'jsx-quotes': 'off',
     'no-duplicate-imports': 'off',
     'no-empty': 'off',
+    'no-useless-return': 'off',
```
Comment (chrisnojima) [inline shared/.eslintrc.js]: conflicts w/ our TS strict rules

File: shared/actions/git.tsx [hunk 2]
```diff
@@ -3,74 +3,84 @@ import * as GregorGen from './gregor-gen'
...
 
-const load = (state: TypedState) =>
-  state.config.loggedIn &&
-  RPCTypes.gitGetAllGitMetadataRpcPromise(undefined, Constants.loadingWaitingKey)
-    .then((results: Array<RPCTypes.GitRepoResult> | null) =>
-      // @ts-ignore codemod-issue
-      GitGen.createLoaded(Constants.parseRepos(results || []))
-    )
-    .catch(() => {})
+const load = async (state: TypedState) => {
```
Comment (chrisnojima) [inline shared/actions/git.tsx]: promise chainActions should mostly be async await

File: shared/constants/entities.tsx [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import * as I from 'immutable'
-import * as Git from './git'
```
Comment (chrisnojima) [inline shared/constants/entities.tsx]: we should mostly deprecate the entity store

File: shared/git/container.tsx [hunk 4]
```diff
@@ -1,134 +1,94 @@
...
-  {} & {
-    _loadGit: (() => void)
-  } & Exclude<
-      React.ElementConfig<typeof Git>,
-      {
-        expandedSet: I.Set<string>
-        onToggleExpand: ((arg0: string) => void)
-      }
-    > */
+const getRepos = memoize((git: Map<string, Types.GitInfo>) =>
```
Comment (chrisnojima) [inline shared/git/container.tsx]: simplify this a bunch

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/git/container.tsx [hunk 5]
```diff
@@ -1,132 +1,94 @@
...
+  sortBy([...git.values()], ['teamname', 'name']).reduce<{personals: Array<string>; teams: Array<string>}>(
```
Comment (MarcoPolo) [inline shared/git/container.tsx]: 🐑 Why did you get rid of partition?
```js
const [teams, personals] = partition(sortBy([...git.values()], ['teamname', 'name']), x => !!x.teamname)
return {teams, personals}
```

Comment (chrisnojima) [reply shared/git/container.tsx]: It still partitions

Comment (MarcoPolo) [reply shared/git/container.tsx]: oh, sorry I missed that you had still call map(x => x.id). this is fine nvm

**Pull request merged**